src/matsim_io/dashboards.py

```diff
@@ -196,68 +196,219 @@ def people_in_safety_csv_path(self) -> Path:
-                            "color": "#444",
-                            "type": "-",
+                    "type": "scatter",
+                    "x": "$dataset.bin",
+                    "y": f"$dataset.cumulative_traveltime_{result.label}",
+                    "name": f"{result.title}",
+                    "original_name": f"{result.title}",
+                    "mode": "lines",
+                    "line": {
+                        "width": 2,
+                        "smoothing": 1,
+                        "shape": "spline",
+                        "dash": "solid",
+                        "simplify": True,
+                        "context": {
+                            "width": 2,
+                            "smoothing": 1,
+                            "shape": "spline",
+                            "dash": "solid",
+                            "simplify": True,
                         },
                     },
                 }
+                for result in results
             ],
-        },
-    }
+            "layout": {
+                "xaxis": {
+                    "title": "Time from start of simulation (minutes)",
+                    "color": "#444",
+                    "type": "-",
+                },
+                "yaxis": {
+                    "title": "Fraction of people in safety",
+                    "color": "#444",
+                    "type": "-",
+                },
+            },
+        }
+    ]
+    return people_in_safety
+
+
+def create_bar_graph_mode_share(
+    results: list[SimulationResult],
+    y: str,
+    x: str,
+    title: str,
+    description: str = "by 10-minute intervals",
+    yaxis_title: str = "Number of people",
+    xaxis_title: str = "Time from start of simulation (minutes)",
+) -> list:  # type: ignore[type-arg]
+    datasets = {}
+    for result in results:
+        datasets[result.label] = {
+            "file": result.mode_share_csv_path,
+            "aggregate": {
+                "func": "SUM",
+                "groupBy": ["dist_group"],
+                "target": "share",
+            },
+        }
 
-    dashboard_path = MATSIM_DATA_DIR / "dashboard-1.yaml"
-    dashboard_path.write_text(yaml.dump(dashboard, sort_keys=False), encoding="utf-8")
+    return _create_bar_graph(
+        results=results,
+        y=y,
+        title=title,
+        x=x,
+        description=description,
+        yaxis_title=yaxis_title,
+        xaxis_title=xaxis_title,
+        datasets=datasets,
+    )
+
+
+def create_bar_graph_trip_purpose(
+    results: list[SimulationResult],
+    y: str,
+    x: str,
+    title: str,
+    description: str = "by 10-minute intervals",
+    yaxis_title: str = "Number of people",
+    xaxis_title: str = "Time from start of simulation (minutes)",
+) -> list:  # type: ignore[type-arg]
+    datasets = {}
+    for result in results:
+        datasets[result.label] = result.trip_purposes_by_10_minutes_csv_path
```

Comment:
If you're curious about other ways to create dictionaries by iteration, then Python also supports dict comprehension, which works similarly to list comprehension:

```suggestion
    datasets = {
        result.label: result.trip_purposes_by_10_minutes_csv_path for result in results
    }
```

Can be neat in some cases, such as in-lining as an argument:

```py
return foo(
    datasets={
        result.label: result.trip_purposes_by_10_minutes_csv_path for result in results
    }
)
```